### DISCUSSION TITLE
--add-data not work for a folder

### DISCUSSION THREAD

**Discussion by BOXERRMD** [pyinstaller/pyinstaller#8282] [Help] [answered]
Hello,
I am developing a small application for Discord and I would like to add a folder in the executable so that the main file can access it.
Unfortunately, every time I run the following command : 

**pyinstaller rpc.py --noconsole --onefile -i circle-3041437_19201.ico --add-data "./Images/*;Images" --add-data "rpc.kv;."**

it doesn't work and returns an error when the main file tries to open a file located, as I've written in the code, at "Image/FILE".

![image](https://github.com/pyinstaller/pyinstaller/assets/130167557/0752245b-ee7f-4cde-a09c-28d2098e98cb)

Could someone please help me find a solution?

**Comment by rokm** [comment, marked as answer]
`with open('Images/save.json', 'r') as file` assumes that this `Images` directory is in your current working directory, which is not the case with onefile builds (nor with onedir ones anymore, for that matter). Anchor the path to `__file__`, as explained in the docs: https://pyinstaller.org/en/stable/runtime-information.html#using-file

---

> `--add-data "./Images/*;Images"`

As a side note, avoid using glob, as it will result in unexpected behavior if `Images` contains sub-directories (i.e., their content will end up directly in `Images` instead of the corresponding sub-directories).

`--add-data "./Images;Images"` should suffice.

**Reply by BOXERRMD** [reply]
Thank you !!!!!! It's work !
